interop/tapir/src/main/scala/caliban/interop/tapir/TapirAdapter.scala

```diff
@@ -309,22 +309,30 @@ object TapirAdapter {
   )(implicit responseCodec: JsonCodec[ResponseValue]): CalibanBody = {
     import DeferMultipart._
 
-    Right(
-      ZStream
-        .unwrapScoped(
-          for {
-            initialAndSubsequent <- stream.peel(ZSink.head[ResponseValue])
-            (initial, subsequent) = initialAndSubsequent
-            payload               = ZStream.fromIterable(
-                                      initial.map(value => resp.copy(data = value).toResponseValue)
-                                    ) ++ subsequent
-          } yield payload
-            .map(responseCodec.encode)
-            .intersperse(InnerBoundary, InnerBoundary, EndBoundary)
+    val pipeline = ZPipeline.fromChannel {
+      lazy val reader: ZChannel[Any, Throwable, Chunk[ResponseValue], Any, Throwable, Chunk[ResponseValue], Any] =
+        ZChannel.readWithCause(
+          (in: Chunk[ResponseValue]) =>
+            in.headOption match {
+              case Some(value) =>
+                ZChannel.write(in.updated(0, resp.copy(data = value).toResponseValue)) *>
+                  ZChannel.identity[Throwable, Chunk[ResponseValue], Any]
+              case None        => reader
```

Comment:
Question for my own understanding: under what conditions would we reach this case? Or is it technically unreachable?